exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java

```diff
@@ -335,8 +336,9 @@ private void javaPropertiesToSystemOptions() {
    * Shutdown hook for Drillbit. Closes the drillbit, and reports on errors that
    * occur during closure, as well as the location the drillbit was started from.
    */
-  private static class ShutdownThread extends Thread {
+  public static class ShutdownThread extends Thread {
     private final static AtomicInteger idCounter = new AtomicInteger(0);
+    private static volatile boolean unrecoverableFailure;
```

Comment:
Consider moving `unrecoverableFailure` to `FailureUtils.java` and keeping `ShutdownThread` as `private`.

Reply:
done